frontend/src/concepts/pipelines/content/import/PipelineVersionImportModal.tsx

```diff
@@ -151,4 +137,8 @@ const PipelineVersionImportModal: React.FC<PipelineVersionImportModalProps> = ({
   );
 };
 
-export default PipelineVersionImportModal;
+const PipelineVersionImportModalWrapper = (props: PipelineVersionImportModalProps) => (
+  <PipelineVersionImportModal key={props.existingPipeline?.id} {...props} />
+);
+
+export default PipelineVersionImportModalWrapper;
```

Comment:
Why is this needed?

Reply:
On the create pipeline runs page, if you change the pipeline selection and click the `Upload new version` button, it will not update the pipeline selection in the pipeline version import modal based on the pipeline selection. This way could guarantee the import versions modal re-renders every time when the pipeline changes.

Reply:
I think this modal is done incorrectly because even if we do not yet open the modal, it has potential for fetching information. We want to avoid doing work for the modal while it it closed. It's likely copying a bad pattern used across the entire app ...

Can we change this modal so that it only does real work when it is opened. I suspect that would solve your issue and remove the need for this `key` trick.

Reply:
Yes, this modal state setting is incorrect across the app lol... Let me see how can I solve this.